blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineRunImpl.java

```diff
@@ -157,6 +159,16 @@ static void clearCache() {
         PIPELINE_NODE_CONTAINER_LOADING_CACHE.invalidateAll();
     }
 
+
+    public static WorkflowRun findRun(String externalizableId) {
+        String[] tree = externalizableId.split("#");
+        Item item = Jenkins.get().getItem(tree[0], Jenkins.get());
+        if (item instanceof WorkflowJob) {
+            return ((WorkflowJob)item).getBuild(tree[1]);
+        }
+        return null;
```

Comment:
please look at the code and you will see the code using this method check if run is null or not.
https://github.com/jenkinsci/blueocean-plugin/blob/65121bccdd759b7fdad6f79c898138820c1a07d4/blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineRunImpl.java#L186
I have added some comments as well..
even the diff show this
<img width="1190" alt="Screenshot 2023-02-15 at 7 17 34 am" src="https://user-images.githubusercontent.com/19728/218865250-c149e33f-4cb3-486d-b347-c0063673f5b7.png">

Reply:
I was referring for example to https://github.com/jenkinsci/blueocean-plugin/blob/65121bccdd759b7fdad6f79c898138820c1a07d4/blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeImpl.java#L180-L187

Reply:
look at the constructor of `PipelineNodeImpl` and btw it's only used by `PipelineNodeGraphVisitor` at this stage if the `WorkflowRun` instance doesn't exist it would have been a NPE long way before.
they cannot be called if the run/build doesn't exist because it's a "sort" of graph tree walk on existing objects. such job -> list of run -> detail of the run -> details of the pipeline steps.
so yes this could fail but it's already the case today if between the time you see the list of runs and then you want to go deeper see detail of a run someone or any event delete a run.

Reply:
Spotbugs is enabled already.
anyway I changed to throw NPE so we can move forward with that.